### PR TITLE
[RLlib] Remove execution plan

### PR DESCRIPTION
**Pull request by avnishn**
Remove the final unused execution plans and now unused execution plan utilities

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/ddppo/ddppo.py [hunk 1]
```diff
@@ -255,31 +255,30 @@ def setup(self, config: PartialTrainerConfigDict):
...
-                    for i, worker in enumerate(self.workers.remote_workers())
-                ]
-            )
-            logger.info("Torch process group init completed")
-            self._ddppo_worker_manager = AsyncRequestsManager(
-                self.workers.remote_workers(),
-                max_remote_requests_in_flight_per_worker=1,
-                ray_wait_timeout_s=0.03,
-            )
+        self._curr_learner_info = {}
```
Comment (gjoliver) [inline rllib/algorithms/ddppo/ddppo.py:258]: can you help comment here that ddppo only has torch implementation, so it's ok to do torch specific work here?

File: rllib/execution/replay_ops.py [hunk 2]
```diff
@@ -1,136 +1,10 @@
-from typing import List, Any, Optional
+from typing import List, Optional
```
Comment (gjoliver) [inline rllib/execution/replay_ops.py:1]: we should be able to delete this whole file?
we should be using Arthur's RBs now right?

File: rllib/execution/train_ops.py [hunk 3]
```diff
@@ -511,21 +511,10 @@ def __call__(
 class UpdateTargetNetwork:
```
Comment (gjoliver) [inline rllib/execution/train_ops.py:511]: writing the comment here, but it's not related to UpdateTargetNetwork.
can we delete the TrainOneStep and MultiGPUTrainOneStep classes now??

**Review by gjoliver — COMMENTED** [review]
awesome! my favorite kind of PR.
thanks a ton man.

**Comment by gjoliver** [comment]
lint and 1 unit test failure.
otherwise, this is great!

**Comment by avnishn** [comment]
ah forgot to remove a test from BUILD that I deleted

**Comment by avnishn** [comment]
ready to merge pending tests passing

**Comment by gjoliver** [comment]
lint / doc error doesn't seem related.

**Review by gjoliver — APPROVED** [review]

File: rllib/execution/replay_ops.py (hunk 2 above)
Comment (ArturNiederfahrenhorst) [reply rllib/execution/replay_ops.py]: Let me get rid of it..

**Comment by sven1977** [comment]
Looks great. Thanks for cleaning this up @avnishn!
There is one problem with the LINTer, it seems:
```
File "../../python/ray/rllib/execution/parallel_requests.py", line ?, in default
--
  | Failed example:
  | manager.call(lambda worker, a, b: worker.task(a, b), fn_args=[1, 2])
  | Exception raised:
  | Traceback (most recent call last):
  | File "/opt/miniconda/lib/python3.6/doctest.py", line 1330, in __run
  | compileflags, 1), test.globs)
  | File "<doctest default[6]>", line 1, in <module>
  | manager.call(lambda worker, a, b: worker.task(a, b), fn_args=[1, 2])
  | NameError: name 'manager' is not defined
```

Maybe we need to add these `# doctest: +SKIP` to the Examples docstring lines?

**Comment by suquark** [comment]
@jiaodong it seems this PR might help your rllib prototyping, since it removes a lot of complexity in the code.

**Comment by jiaodong** [comment]
@suquark thanks for ccing me here :) 2000 less lines of code sounds great indeed

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Great stuff. I checked out the PR and found 1-2 minor occurences of "execution_plan" and changed them myself. When are we going to do MBMPO and MAML? Anything keeping us from writing training iteration functions on these?

**Comment by sven1977** [comment]
Yeah, we still need to complete MB-MPO and MAML. We left them out for no specific reasons.

**Comment by ArturNiederfahrenhorst** [comment]
Let's put them on my next sprint, please.

**Comment by mattip** [comment]
This PR removed rllib/execution/concurrency_ops.py and rllib/execution/multi_gpu_impl.py which are referred to in the documentation, so the [book-lint CI run](https://buildkite.com/ray-project/ray-builders-pr/builds/35469#01816327-6b20-4927-acad-a3ae360e3096/1455-3169) is failing. [here](https://github.com/ray-project/ray/blob/master/doc/source/rllib/core-concepts.rst) is the documentation source with the file names.

**Comment by mattip** [comment]
[here](https://github.com/ray-project/ray/blob/57d02eec2e369d659c73b758586306a30e687cd8/doc/source/rllib/core-concepts.rst?plain=1#L282) is the exact reference to `concurrency_ops.py` and [here](https://github.com/ray-project/ray/blob/57d02eec2e369d659c73b758586306a30e687cd8/doc/source/rllib/rllib-models.rst?plain=1#L362) is the reference to `multi_gpu_impl.py`

**Pull request merged**